src/features/collections/data_products/TaxaCount.tsx

```diff
@@ -25,15 +26,28 @@ export const TaxaCount: FC<{
   }, [ranksQuery.data]);
 
   // Counts
+  const matchId = useAppParam('match');
   const countsParams = useMemo(
-    () => ({ collection_id, rank: String(rank?.value) }),
-    [collection_id, rank?.value]
+    () => ({ collection_id, rank: String(rank?.value), match_id: matchId }),
+    [collection_id, matchId, rank?.value]
   );
-  const countsQuery = getTaxaCountRank.useQuery(countsParams, { skip: !rank });
-  const counts = countsQuery.data?.data || [];
-  const max = counts.reduce((max, { count }) => (max > count ? max : count), 0);
+  const [polling, setPolling] = useState(false);
+  useEffect(() => setPolling(!!countsParams.match_id), [countsParams]); // Poll for new queries with a matchId
```

Comment:
We should probably decide on a preferred style (operators versus explicit type casts) for things like
- `Boolean(val)`
- `!!val`
- `Number(val)`
- `+val`